micrometer-core/src/main/java/io/micrometer/core/instrument/binder/jpa/HibernateMetrics.java

```diff
@@ -250,4 +264,97 @@ private SessionFactory unwrap(EntityManagerFactory entityManagerFactory) {
+                                .register(meterRegistry);
+
+                        FunctionCounter.builder("hibernate.query.cache.requests", queryStatistics, QueryStatistics::getCacheMissCount)
+                                .tags(tags)
+                                .tags("result", "miss", "query", queryName)
+                                .description("The number of query cache miss")
+                                .register(meterRegistry);
+
+                        FunctionCounter.builder("hibernate.query.cache.puts", queryStatistics, QueryStatistics::getCacheMissCount)
+                                .tags(tags)
+                                .tags("query", queryName)
+                                .description("The number of putting the query into cache")
+                                .register(meterRegistry);
+
+                        FunctionCounter.builder("hibernate.query.executions", queryStatistics, QueryStatistics::getExecutionCount)
+                                .tags(tags)
+                                .tags("query", queryName)
+                                .description("Total number of query executions")
+                                .register(meterRegistry);
+
+                        TimeGauge.builder("hibernate.query.executions.total", queryStatistics, TimeUnit.MILLISECONDS, QueryStatistics::getExecutionTotalTime)
+                                .tags(tags)
+                                .tags("query", queryName)
+                                .description("Query total execution time")
+                                .register(meterRegistry);
+
+                        TimeGauge.builder("hibernate.query.executions.avg", queryStatistics, TimeUnit.MILLISECONDS, QueryStatistics::getExecutionAvgTime)
+                                .tags(tags)
+                                .tags("query", queryName)
+                                .description("Query average execution time")
+                                .register(meterRegistry);
+
+                        TimeGauge.builder("hibernate.query.executions.max", queryStatistics, TimeUnit.MILLISECONDS, QueryStatistics::getExecutionMaxTime)
+                                .tags(tags)
+                                .tags("query", queryName)
+                                .description("Query maximum execution time")
+                                .register(meterRegistry);
+
+                        TimeGauge.builder("hibernate.query.executions.min", queryStatistics, TimeUnit.MILLISECONDS, QueryStatistics::getExecutionMinTime)
+                                .tags(tags)
+                                .tags("query", queryName)
+                                .description("Query minimum execution time")
+                                .register(meterRegistry);
+
+                        FunctionCounter.builder("hibernate.query.executions.rows", queryStatistics, QueryStatistics::getExecutionRowCount)
```

Comment:
Is this a counter (meaning this number only goes up)? Or is it a gauge of the latest rows returned by a query?

Reply:
Yes, this is the total cumulatively increasing number of rows returned by specific query, so I think counter is ok